interp/constrexpr.ml

```diff
@@ -20,7 +20,10 @@ type universe_decl_expr = (lident list, Glob_term.glob_constraint list) UState.g
 type ident_decl = lident * universe_decl_expr option
 type name_decl = lname * universe_decl_expr option
 
-type notation = string
+type notation_entry = InConstrEntry | InCustomEntry of string
```

Comment:
Why is `constr` special? IMHO, we could benefit from a more generic handling here.

Reply:
Currently, `constr` remains special. For instance, insertion of parentheses at printing time could now be done generically as it is done for custom entry (in particular one could eventually get rid of the `if prec_less` in `ppconstr.ml`). The special treatment of `{` ... `}` is also only for `constr`.

Eventually, the code could not have to distinguish between `InConstrEntry` and `InCustomEntry` (or between `InConstrEntrySomeLevel` and `InCustomEntryLevel`), but that's not yet the case.